models/middleware.js

```diff
@@ -0,0 +1,296 @@
+var app = require('../app');
+var loopback = require('loopback');
+var debug = require('debug')('workspace:middleware');
+var ConfigFile = app.models.ConfigFile;
+var path = require('path');
+var stableSortInPlace = require('stable').inplace;
+
+/**
+ * Defines a `Middleware` configuration.
+ * @class Middleware
+ * @inherits Definition
+ */
+
+var Middleware = app.models.Middleware;
+
+/**
+ * - `facetName` is required and must refer to an existing facet
+ *
+ * @header Property Validation
+ */
+Middleware.validatesPresenceOf('facetName');
+
+function getPhase(name) {
+  if (name == null) {
+    return name;
+  }
+  var parts = name.split(':');
+  return parts[0];
+}
+
+/**
+ * Get all phases by name
+ * @param {Function} cb Callback function
+ */
+Middleware.getPhases = function(cb) {
+  // Load all entries from cache and sort them by order
+  Middleware.find(function(err, instances) {
+    if (err) return cb(err);
+    var phases = sortMiddleware(instances).phases;
+    cb(null, phases);
+  });
+}
+
+var ORDER_BUFFER = 1024;
+
+/**
+ * Add a middleware phase to the cache in a slot after the `nextPhase`
+ * @param {String} facetName
+ * @param {String} phase
+ * @param {String} nextPhase
+ */
+Middleware.addPhase = function(facetName, phase, nextPhase, cb) {
```

Comment:
Ditto, I think this method should be implemented by `MiddlewarePhase`. What are the benefits of keeping it here, in `Middleware`?